#!/usr/bin/env python
PACKAGE = "mantis_state"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

gen.add("flight_ready", bool_t, 0, "Toggle to set the system as flight ready", False)

gen.add("px", double_t, 0, "X Position", 0.0)
gen.add("py", double_t, 0, "Y Position", 0.0)
gen.add("pz", double_t, 0, "Z Position", 1.0)

gen.add("r1", double_t, 0, "Joint 1 Position", 0.0)
gen.add("r2", double_t, 0, "Joint 2 Position", 0.0)
gen.add("r3", double_t, 0, "Joint 3 Position", 0.0)
gen.add("r4", double_t, 0, "Joint 4 Position", 0.0)
gen.add("r5", double_t, 0, "Joint 5 Position", 0.0)
gen.add("r6", double_t, 0, "Joint 6 Position", 0.0)
gen.add("r7", double_t, 0, "Joint 7 Position", 0.0)
gen.add("r8", double_t, 0, "Joint 8 Position", 0.0)

exit(gen.generate(PACKAGE, "mantis_state_xacro_test_node", "XacroTest"))
